### PR TITLE
[DEPENDS ON #2458] [Java] Add DataType Timex Resolver Library Unit Tests

### PR DESCRIPTION
**Pull request by Batta32**
Depends on #2458

## Description
The DataType Timex Resolver library is missing in Java. This PR contains the migrated library from C# to Java (only tests).

## Specific Changes
- Add `TestTime` unit tests
- Add `TestTimex` unit tests
- Add `TestTimexConvert` unit tests
- Add `TestTimexCreator` unit tests
- Add `TestTimexDateHelpers` unit tests
- Add `TestTimexFormat` unit tests
- Add `TestTimexHelpers` unit tests
- Add `TestTimexParsing` unit tests
- Add `TestTimexRangeResolve` unit tests
- Add `TestTimexRelativeConvert` unit tests
- Add `TestTimexResolver` unit tests

## Testing
All the tests are correctly passing 281/281. These tests were validated using JDK 8/11/15

_Unit tests passing correctly_
image

_Main structure of the migrated tests_
image

### CODE REVIEW

**Review by tellarin — CHANGES_REQUESTED** [review]
Just marking it to hold merge until #2458 is in.

**Comment by tellarin** [comment]
@Batta32, could you update this PR to include only the relevant changes?

**Comment by Batta32** [comment]
@tellarin - we updated the branch with the last changes of `master` in order to have only the relevant changes 😊.

**Review by tellarin — CHANGES_REQUESTED** [review]
Overall it looks good, but it seems to be missing some unit test cases (at least some from the latests PRs fixing issues in the .NET version of the library).

**Comment by Batta32** [comment]
@tellarin - as you mentioned, we incorporated the fixes of PR #2469. We double checked the unit tests and it's in parity with C#.

File: Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexRangeResolve.java [hunk 1]
```diff
@@ -0,0 +1,1099 @@
...
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestTimexRangeResolve {
```
Comment (tellarin) [inline Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexRangeResolve.java:20]: Minor: Should be "TestTimexRangeResolver".

**Review by tellarin — CHANGES_REQUESTED** [review]
Sorry, my mistake. I think I looked at the wrong place and didn't see the new tests.
There is just a minor naming issues I hadn't seen before.
Thanks.

File: Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexRangeResolve.java (hunk 1 above)
Comment (Batta32) [reply Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexRangeResolve.java]: Done!

**Comment by Batta32** [comment]
@tellarin, we applied the requested change. Just FYI, in C# the file is [TestTimexRangeResolve](https://github.com/microsoft/Recognizers-Text/blob/master/.NET/Microsoft.Recognizers.Text.DataTypes.DataDrivenTests/TestTimexRangeResolve.cs).

**Comment by tellarin** [comment]
> @tellarin, we applied the requested change. Just FYI, in C# the file is [TestTimexRangeResolve](https://github.com/microsoft/Recognizers-Text/blob/master/.NET/Microsoft.Recognizers.Text.DataTypes.DataDrivenTests/TestTimexRangeResolve.cs).

Didn't notice that. Could you also rename it?

**Comment by Batta32** [comment]
@tellarin, done!

**Review by tellarin — APPROVED** [review]

**Pull request merged**